projects/library/src/lib/states/primitive.state.ts

```diff
@@ -4,4 +4,6 @@ import { BaseState } from './base.state';
  *	@class
  *	@classdes Primitive state class. Used for save state with Primitive type (Like boolean, number or string).
  */
-export class PrimitiveState<T> extends BaseState<T> {}
+export class PrimitiveState<
+  T = string | number | boolean | bigint | symbol
```

Comment:
This approach isn't flexible. What if I want to use some concrete `type` or `key of enum` as the primitive?

Like:
```
enum MyEnum {
   Value1,
   Value2
}

type MyType = 'male' | 'female';

PrimitiveState<MyEnum>
PrimitiveState<MyType>
```

Reply:
how about using ObjectState for such cases and leaving primitive for explicit primitives?

Reply:
or, in order to cover all areas, make EnumState and TypeState

Reply:
although everything is correct, set is typed T and maybe my PR is overkill

Reply:
on the other hand without T = string | number | boolean | bigint | symbol nothing prevents us from using PrimitiveState (and other implementations of BaseState) with any type T, for example 
``` export class SelectedUser extends PrimitiveState<{name: 'name', last:'last'}> {} ```